### PR TITLE
Initial implementation of DQ step code.

### PR DESCRIPTION
**Pull request by PaulHuwe**
This is my first attempt at implementing and integrating, so I am asking for a pre-review of my draft PR. 

Please note the temporarily commented out sections for Guiding models, as those have yet to be implemented, but will be 
(hence not removing that code entirely). 

In addition, can you confirm that the contents of guider_list are correct?

### CODE REVIEW

File: romancal/dq_init/dq_initialization.py [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+from .. import datamodels
+#from ..lib import reffile_utils
+
+log = logging.getLogger(__name__)
+log.setLevel(logging.DEBUG)
+
+
+# Guide star mode exposure types
+guider_list = ['WFI_WIM_ACQ','WFI_WIM_TRACK','WFI_WSM_ACQ1','WFI_WSM_ACQ2','WFI_WSM_TRACK']
+
```
Comment (ddavis-stsci) [inline romancal/dq_init/dq_initialization.py:14]: These modes look fine to me and match the guide window table 3.5.2.1 Table of Guide Window Sizes and Use.

File: romancal/dq_init/dq_initialization.py [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+    # if reffile_utils.ref_matches_sci(output_model, mask_model):
+    #     mask_array = mask_model.dq
+    # else:
+    #     log.info('Extracting mask subarray to match science data')
+    #     mask_sub_model = reffile_utils.get_subarray_model(output_model,
+    #                                                       mask_model)
+    #     mask_array = mask_sub_model.dq.copy()
+    #     mask_sub_model.close()
+
+    # Extract subarray from reference data, if necessary
```
Comment (ddavis-stsci) [inline romancal/dq_init/dq_initialization.py:73]: I think you can just remove the subarray code.

**Review by ddavis-stsci — APPROVED** [review]

File: romancal/dq_init/dq_initialization.py (hunk 1 above)
Comment (dmggh) [reply romancal/dq_init/dq_initialization.py:14]: Is the intermediate function correct_model() needed ? It consists of a single statement calling another function having the same arguments. Perhaps I'm missing something here.

File: romancal/dq_init/dq_initialization.py [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+
+log = logging.getLogger(__name__)
+log.setLevel(logging.DEBUG)
+
+
+# Guide star mode exposure types
+guider_list = ['WFI_WIM_ACQ','WFI_WIM_TRACK','WFI_WSM_ACQ1','WFI_WSM_ACQ2','WFI_WSM_TRACK']
+
+
+def correct_model(input_model, mask_model):
```
Comment (nden) [inline romancal/dq_init/dq_initialization.py:16]: This function seems unnecessary.

File: romancal/dq_init/dq_initialization.py [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+    ----------
+    input_model : input Roman datamodel
+        The Roman datamodel to be corrected
+
+    mask_model : mask datamodel
+        The mask model to use in the correction
+
+    Returns
+    -------
+    output_model : Roman datamodel
```
Comment (nden) [inline romancal/dq_init/dq_initialization.py:51]: What model is this?

File: romancal/dq_init/dq_initialization.py [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+
+def do_dqinit(input_model, mask_model):
+    """Perform the dq_init step on a Roman datamodel
+
+    Parameters
+    ----------
+    input_model : input Roman datamodel
+        The Roman datamodel to be corrected
+
+    mask_model : mask datamodel
```
Comment (nden) [inline romancal/dq_init/dq_initialization.py:46]: Does linking work, i.e. `rdm.MaskModel`?

File: romancal/dq_init/dq_initialization.py (hunk 1 above)
Comment (nden) [reply romancal/dq_init/dq_initialization.py:14]: The code in `do_dqinit` assumes these values are enumerated in `meta.exposure.type` .

File: romancal/dq_init/dq_initialization.py [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+    -------
+    output_model : Roman datamodel
+        The corrected Roman datamodel
+    """
+
+    # Inflate empty DQ array, if necessary
+    check_dimensions(input_model)
+
+    # Create output model as copy of input
+    output_model = input_model.copy()
```
Comment (nden) [inline romancal/dq_init/dq_initialization.py:59]: We should check carefully if a copy is really needed here.

**Comment by PaulHuwe** [comment]
Deprecated in favor of #252

**Pull request closed without merging**